api/v1alpha1/toolchaincluster_types.go

```diff
@@ -66,6 +79,13 @@ type LocalSecretReference struct {
 // cluster updated periodically by cluster controller.
 // +k8s:openapi-gen=true
 type ToolchainClusterStatus struct {
+	// APIEndpoint is the API endpoint of the remote cluster. This can be a hostname,
+	// hostname:port, IP or IP:port.
+	APIEndpoint string `json:"apiEndpoint"`
+
+	// OperatorNamespace is the namespace in which the operator runs in the remote cluster
+	OperatorNamespace string `json:"operatorNamespace"`
```

Comment:
You need to make these fields optional - there are three reasons:
1. if you kept it as it is, then all already existing ToolchainCluster CRs would become invalid (as they don't have the fields set). I'm not user if OLM would be able to update the CRD at all.
2. ToolchainCluster controller doesn't have the logic that populates these fields yet, so it would not be able to update the ToolchainClusterStatus at all (rejected as the fields are required) until the logic is added.
3. In the future, if the ToolchainCluster is provided with either non-existing or invalid secret/kubeconfig, then we would need to set the "ready" condition to false and leave these two fields unset (empty)
```suggestion
	// hostname:port, IP or IP:port.
	// +optional
	APIEndpoint string `json:"apiEndpoint,omitempty"`

	// OperatorNamespace is the namespace in which the operator runs in the remote cluster
	// +optional
	OperatorNamespace string `json:"operatorNamespace,omitempty"`
```

Reply:
Just looking at the e2e tests in the other PRs and yes, they both failed in the migration test suite:
```
  - lastTransitionTime: "2024-07-04T19:47:13Z"
    message: 'error validating existing CRs against new CRD''s schema for "toolchainclusters.toolchain.dev.openshift.com":
      error validating toolchain.dev.openshift.com/v1alpha1, Kind=ToolchainCluster
      "toolchain-host-04194050/member-ci-op-80dsm6yr-1de72.devsandboxci.devcluster.openshift.1":
      updated validation is too restrictive: [[].status.apiEndpoint: Required value,
      [].status.operatorNamespace: Required value]'
    reason: InstallComponentFailed
    status: "True"
    type: InstallPlanFailed
```

Reply:
Ohhh jeez, I haven't thought about this yesterday when reviewing the PR - that we cannot introduce required fields into existing CRs 🤦‍♂️ . Great point @MatousJobanek !

Reply:
Yeah, I was suspecting that but wasn't 100% how k8s would behave, so I tried. CI seems to be failing with OLM not being able to upgrade.